presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/glue/GlueHiveMetastore.java

```diff
@@ -643,6 +655,37 @@ public void dropColumn(String databaseName, String tableName, String columnName)
     }
 
     private List<Partition> getPartitions(String databaseName, String tableName, String expression)
+    {
+        if (numSegments == 1) {
+            return getPartitions(databaseName, tableName, expression, null);
+        }
+
+        // Do parallel partition fetch.
+        CompletionService<List<Partition>> completionService = new ExecutorCompletionService<>(executor);
+        List<Segment> segments = IntStream.range(0, numSegments)
+                .mapToObj(s -> new Segment().withSegmentNumber(s).withTotalSegments(numSegments))
+                .collect(toList());
+        for (Segment segment : segments) {
+            completionService.submit(() -> getPartitions(databaseName, tableName, expression, segment));
+        }
+
+        try {
+            List<Partition> partitions = new ArrayList<>();
+            for (int i = 0; i < segments.size(); i++) {
+                Future<List<Partition>> futurePartitions = completionService.take();
+                partitions.addAll(futurePartitions.get());
+            }
+            return partitions;
```

Comment:
Sort partitions before returning. We want planning to be deterministic (as much as possible).

Reply:
When sorting, please do this outside of try block, so that try-catch encompasses as little as possible.

Reply:
OK. Will do the sorting based on partition values.